src/Compiler/Optimize/LowerComputedCollections.fs

```diff
@@ -252,15 +252,77 @@ let (|SeqToArray|_|) g expr =
     | ValApp g g.seq_to_array_vref (_, [seqExpr], m) -> Some (seqExpr, m)
     | _ -> None
 
+[<return: Struct>]
+let (|ConstInt32Range|_|) g expr =
```

Comment:
Could you please add code-samples-as-comments for the active patterns here as well? It makes reading easier.

Reply:
dcc970d